lib/application.js

```diff
@@ -471,15 +539,47 @@ var RequireAuthorizationApplication = Application.extend({
             });
         });
     },
-    getAccessToken: function(token, secret, verifier, callback, options) {
+    setAccessToken: function(token, secret, verifier, callback, options) {
         var self = this;
 
         return new Promise(function(resolve, reject) {
-            self.oa.getOAuthAccessToken(token, secret, verifier, function(err, token, secret, results) {
+            self.oa.getOAuthAccessToken(token, secret, verifier, function(err, results) {
                 if (err)
                     reject(err);
-                else
-                    resolve({ token: token, secret: secret, results: results });
+                else {
+                    var exp = new Date();
+                    exp.setTime(exp.getTime() + (results.oauth_expires_in * 1000));
+                    self.setOptions({
+                        accessToken: results.oauth_token,
+                        accessSecret: results.oauth_token_secret,
+                        sessionHandle: results.oauth_session_handle,
+                        tokenExpiry: exp.toString()
+                    });
+                    resolve({ results: results });
```

Comment:
I suppose it depends on how the 'client' object is persisted.  If this is planned to be created on each 'run' of the app, then we'll need to have a constructor that's capable of recreating a client with an access token as a parameter.

I'm not sure if it's capable of this at the moment.